lib/Model/Pass/EnsureAllFunctionsAreNamed.cpp

```diff
@@ -0,0 +1,53 @@
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_set>
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Pass/RegisterModelPass.h"
+
+struct TransparentHash {
+  using is_transparent = void;
+  using transparent_key_equal = std::equal_to<std::string_view>;
+  size_t operator()(std::string_view View) const {
+    return std::hash<std::string_view>{}(View);
+  }
+  size_t operator()(llvm::StringRef View) const {
+    return operator()(std::string_view{ View });
+  }
+};
+using NameSet = std::unordered_set<model::Identifier, TransparentHash>;
+
+static void ensureAllFunctionsAreNamed(TupleTree<model::Binary> &Model) {
```

Comment:
Somehow I knew it was going to be the case 😅.

First of all, here's the original thread where we discussed this: https://github.com/revng/revng/pull/232#discussion_r1054659008

We decided to rely onto the `CustomName`s instead of `ID`s to differentiate function types. Except, most of the function types have no name, hence it forces me to invent my own for the sake of this test.

P. S. This could also be a tool in `tests/abi/tools` or something, I don't mind either way.

P. P. S. There's also an option of forcing `dwarf` to write the names of function types (and not only functions), but I'd imagine there are non-trivial implications in there as well.